core/src/server/rpc_module.rs

```diff
@@ -891,14 +892,26 @@ impl SubscriptionSink {
 		T: Serialize,
 		E: std::fmt::Display,
 	{
-		let close_notify = match self.close_notify.clone() {
+		let conn_closed = match self.close_notify.clone() {
 			Some(close_notify) => close_notify,
-			None => return SubscriptionClosed::RemotePeerAborted,
+			None => {
+				return SubscriptionClosed::RemotePeerAborted;
+			}
+		};
+
+		let sub_closed = match self.unsubscribe.take() {
```

Comment:
Ah I see, so this oneshot unsubscribe channel is removed here, which means we can only use `pipe_from_try_stream` once.

Is it possible to take a mutable ref to it instead to use below (which may need pinning)? Maybe then this method would still be reusable?

Reply:
it won't work because `send` takes a `&self as well.

I changed it to `tokio::sync::watch` for ease of use.